Python/crossfire/crossfire/occurrences.py

```diff
@@ -0,0 +1,32 @@
+from queue import Queue, Empty
+
+
+class Occurrences:
+    def __init__(self, client):
+        self.client = client
+        self.buffer = Queue()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.buffer.empty():
+            self.load_occurrences()
+
+        try:
+            occurrences = self.buffer.get_nowait()
+        except Empty:
+            raise StopIteration
+
+        return occurrences
+
+    def load_occurrences(self):
+        occurrences = self.client.get(
+            "https://api-service.fogocruzado.org.br/api/v2/occurrences?"
+        )
+        if occurrences.get("pageMeta", {}).get("hasNextPage"):
```

Comment:
… isso vai ser _sempre_ `None`, pois `self.client.get` retorna apenas as ocorrências (não o `pageMeta`)

Reply:
Fiz [um examplo](https://github.com/FelipeSBarros/crossfire/compare/master...cuducos:crossfire:decorator-with-pagination). O que acha?

Reply:
Acho que podemos fazer o decorador receber um `pagination=True|False` e se for `False` ele só retorna os dados…

Aí podemos mover ele para os métodos:

```python
@parse_response(pagination=False)
def states(self, *args, **kwargs): # this returns only data
    ...

@parse_response(pagination=True)
def occurences(self, *args, **kwargs): # this returns data and the boolean telling if there's a next page
    ...
```

Reply:
Hum… não. Isso mudaria muito o como lidamos com o `format`… 
Podemos abandonar o decorador e só chamar ele como função mesmo…
Enfim. 

Vou esperar ovir de ti o que acha dessas sugestões : )

Reply:
> Fiz [um exemplo](https://github.com/FelipeSBarros/crossfire/compare/master...cuducos:crossfire:decorator-with-pagination). O que acha?

Show. Fiz um branch e um PR, na master...

Reply:
> Acho que podemos fazer o decorador receber um `pagination=True|False` e se for `False` ele só retorna os dados…

OK. Se entendi bem, além de usarmos o `parser_response` retornando tupla (datos e info e paginação) podemos aidionar ao decorador um parâmetro `pagination` que vai definir se retorna dado e paginação ou apenas dado.
Confere?

Reply:
> ```python
> @parse_response(pagination=False)
> def states(self, *args, **kwargs): # this returns only data
>     ...
> 
> @parse_response(pagination=True)
> def occurences(self, *args, **kwargs): # this returns data and the boolean telling if there's a next page
>     ...
> ```

Mas não é o método `get()` que recebe o  decorador?
Seria o caso de mudar?
Fiquei meio perdido...

Reply:
> Seria o caso de mudar?

Sim por isso mencionei:

> Aí podemos mover ele para os métodos

Reply:
> Fiz um branch e um PR, na master...
 
 Qual o número do PR? Linka aqui pra gente.
 Quer que eu abra um PR com a minha branch?

Reply:
> Qual o número do PR? Linka aqui pra gente.

Editei o comentário com o link ao PR #34